packages/components/loading/src/directive.ts

```diff
@@ -1,56 +1,100 @@
-import Loading from './index'
+import { isRef, ref } from 'vue'
+import { isObject, isString } from '@vue/shared'
+import { Loading } from './service'
+import type { Directive, DirectiveBinding, UnwrapRef } from 'vue'
+import type { LoadingOptions } from './types'
+import type { LoadingInstance } from './loading'
 
-import type { DirectiveBinding } from 'vue'
-import type { ILoadingInstance } from './loading.type'
-
-const INSTANCE_NAME = 'ElLoading'
+const INSTANCE_KEY = Symbol('ElLoading')
 
+export type LoadingBinding = boolean | UnwrapRef<LoadingOptions>
 export interface ElementLoading extends HTMLElement {
-  [INSTANCE_NAME]?: ILoadingInstance
+  [INSTANCE_KEY]?: {
+    instance: LoadingInstance
+    options: LoadingOptions
+  }
 }
 
-const createInstance = (el: ElementLoading, binding: DirectiveBinding) => {
-  const textExr = el.getAttribute('element-loading-text')
-  const spinnerExr = el.getAttribute('element-loading-spinner')
-  const svgExr = el.getAttribute('element-loading-svg')
-  const svgViewBoxExr = el.getAttribute('element-loading-svg-view-box')
-  const backgroundExr = el.getAttribute('element-loading-background')
-  const customClassExr = el.getAttribute('element-loading-custom-class')
+const createInstance = (
+  el: ElementLoading,
+  binding: DirectiveBinding<LoadingBinding>
+) => {
   const vm = binding.instance
-  el[INSTANCE_NAME] = Loading({
-    text: (vm && vm[textExr]) || textExr,
-    svg: (vm && vm[svgExr]) || svgExr,
-    svgViewBox: (vm && vm[svgViewBoxExr]) || svgViewBoxExr,
-    spinner: (vm && vm[spinnerExr]) || spinnerExr,
-    background: (vm && vm[backgroundExr]) || backgroundExr,
-    customClass: (vm && vm[customClassExr]) || customClassExr,
-    fullscreen: !!binding.modifiers.fullscreen,
-    target: binding.modifiers.fullscreen ? null : el,
-    body: !!binding.modifiers.body,
-    visible: true,
-    lock: !!binding.modifiers.lock,
-  })
+
+  const getBindingProp = <K extends keyof LoadingOptions>(
+    key: K
+  ): LoadingOptions[K] =>
+    isObject(binding.value) ? binding.value[key] : undefined
+  const resolveExpression = (key: any) => {
+    const data = (isString(key) && vm?.[key]) || key
+    if (data) return ref(data)
+    else return data
+  }
+
+  const text = getBindingProp('text') || el.getAttribute('element-loading-text')
```

Comment:
**Suggestion:** encapsulate these into function:

```ts
const getProp = (name: string) => {
  return getBindingProp(name) || el.getAttribute(`element-loading-${hyphen(name)}`)
}
```

Reply:
1. `svgViewBox` -> `element-loading-svg-view-box` not same.
2. I planed to remove `element-loading-xxx` in the next minor release. So let's keep it as is.